Content.Tests/DMProject/Tests/Preprocessor/macro_newline_var.dm

```diff
@@ -0,0 +1,51 @@
+#define DEFINE_FLOORS(_PATH, _VARS) \
+	/obj/simulated/floor/_PATH{_VARS};\
+	/obj/unsimulated/floor/_PATH{_VARS};\
+	/obj/simulated/floor/airless/_PATH{_VARS};\
+	/obj/unsimulated/floor/airless/_PATH{_VARS};
+
+var/list/gvars_datum_init_order = list()
+
+#define GLOBAL_RAW(X) /var/global##X
+#define GLOBAL_MANAGED(X, InitValue)\
+/proc/InitGlobal##X(){\
+	##X = ##InitValue;\
+	gvars_datum_init_order += #X;\
+}
+#define GLOBAL_LIST_INIT(X, InitValue) GLOBAL_RAW(/list/##X); GLOBAL_MANAGED(X, InitValue)
+
+DEFINE_FLOORS(carpet/regalcarpet,
+	name = "regal carpet";\
+	icon = 'icons/obj/floors.dmi';\
+	icon_state = "regal_carpet";)
+
+DEFINE_FLOORS(carpet/regalcarpet/border,
+	name = "regal carpet border";\
+	icon = 'icons/obj/floors.dmi';\
```

Comment:
i mean, we don't have that icon file right? probably should change the tests slightly

Reply:
fair point, although shouldn't we be erroring on that?